distributed/diagnostics/tests/test_scheduler_plugin.py

```diff
@@ -125,3 +125,16 @@ async def close(self):
 
     assert plugin.history == ["started", "closed"]
     assert plugin.scheduler is s
+
+
+@gen_cluster(client=True)
+async def test_register_scheduler_plugin(c, s, a, b):
+    temp_list = []
+
+    class Dummy(SchedulerPlugin):
+        def start(self, scheduler):
+            temp_list.append(1)
+
+    await c._register_scheduler_plugin(Dummy)
```

Comment:
The use `self.sync` in `Client.register_scheduler_plugin` will allow us to use it in both sync and async contexts:

```suggestion
    await c.register_scheduler_plugin(Dummy)
```